### ISSUE TITLE
Error in process of using DeepSparse for quantized ONNX model

### ISSUE THREAD

**Issue by Powss** [neuralmagic/deepsparse#1674]
**Describe the bug**
I used the UNet segmentation model and optimization via sparseml (quantization aware training or QAT) and everything is fine during quantization, but when I use the compile_model method from deepsparse for such an ONNX model, deepsparse gives the following error:

Date: 04-11-2025 @ 06:03:54 UTC
OS: Linux teslav100 5.15.0-122-generic #132~20.04.1-Ubuntu SMP Fri Aug 30 15:50:07 UTC 2024
Arch: x86_64
CPU: GenuineIntel
Vendor: Intel
Cores/sockets/threads: [16, 2, 32]
Available cores/sockets/threads: [16, 2, 32]
L1 cache size data/instruction: 32k/32k
L2 cache size: 0.25Mb
L3 cache size: 20Mb
Total memory: 259.79G
Free memory: 35.8898G
Thread: 0x7fa3d4cfd740

Assertion at ./src/include/wand/jit/function/fusors/scalar_elementwise.hpp:43

Backtrace:
 0# 0x00007fa232e5bc63: 
    [440fb6c34c8b25b24585006a004489e94c89f641b9010000004c89e7e8bc887d]
    [00585a84db75084c89e7e89ea2feff4c89e7e8b6887d0048833dae4085000074]
 1# 0x00007fa232f1da4f: 
    [9df1f3ffb901000000ba2b000000488d35e4b0eafc488d3df52edcfce8c0e1f3]
    [ff4989c4eb0a4889c5eb284989c4eb0b4889efc5f877e84638f9ffbee8030000]
 2# 0x00007fa232e69bc5: 
    [20c5fd7f4424603cff741f4889ca4989fc488d0d434383004889e6c5f877ff14]
    [c14c89e04c8b65f8c9c3488d3dda89f3fcc5f877e880e762fdcccc4154555348]
 3# 0x00007fa232ea3053: 
    [4c8d7424200f1f8400000000004d89f04889d94c89e24889ee4c89ffe8ec6afc]
    [ff488b8424400300004c89efc5fa6f8c2420030000c5fa6f942430030000c5fa]
 4# 0x00007fa232f20713: 
    [48020000e824f9ffff488d5c2420488b5424084c89f14c89e64889dfe88c25f8]
    [ff488b542430c5f96f6c24204c89ef48c744243000000000498b842428030000]
 5# 0x00007fa232a72ffd: 
    [e8fe90fbff488b7424184531c04c89e941b9010000004c89fa4c89f7e802d54a]
    [00488bbc24480100004885ff7405e87060a0fd4c8d742440488b5424304c89e6]
 6# 0x00007fa232a73dd5: 
    [00c4c17a7fa424100200004c8da550fdffff4c89e648898558fdffffe8aaf0ff]
    [ff488b9d58fdffff4c8ba550fdffff4c39e374400f1f8000000000498bbc2400]
 7# 0x00007fa231aaaeff: 
    [f2ffff488b8dc8f2ffffc68500f3ffff0048c785f8f2ffff00000000e85088fc]
    [00584c89e75ae876f97401488985c8f2ffff488b8568f5ffff4c89e7488985c0]
 8# 0x00007fa233082a62: 
    [488d4e20488d56104889e74c8d8e600200004c8d86480200004c89d6e8cd83a2]
    [fec5f96f4c2410488b7c2418488b0c24488b442408488b542410c4c17a7f0c24]
 9# 0x00007fa233081cd7: 
    [00f00fc11085d20f85a400000048837b10000f84f70000004889e74889deff53]
    [18c5f96f0424488b7b28c5f1efc9c5f97f0c24c5fa7f43204885ff0f84880000]
10# 0x00007fa231d1ce50: 
    [4883ec18488b07488b16488378100048895424087410488d7424084889c7ff50]
    [184883c418c3e8773d8d01ccccccccccb801000000c3488b45104885c07425ba]
11# 0x00007fa231d340c5: 
    [89442410488b4424084889442418744b488d542418488d7424104c89f741ff56]
    [18498b4e30498b56284839ca741e0f1f440000488b0248c1e00649034778f048]
12# 0x00007fa231d34603: 
    [415d415e415fc3660f1f440000488b0c24488b5424084889de4889efe83cfaff]
    [ffe9dffeffff0f1f80000000004883c201488d7d1048895558e8ff94feff4883]
13# 0x00007fa231d306ad: 
    [0f1f00f3908b0139c274f849837d1000740f4c897c24084c89f64c89ef41ff55]
    [18488b83d800000041b8010000008b9080000000488d8880000000488d7040f0]
14# 0x00007fa231d2c08e: 
    [1000488b09488b12488b14ca488954240874134889f0488d7424084889c7ff50]
    [184883c418c3e8394b8c01cccccccccccccc48c7471000000000488b4a104889]
15# 0x00007fa231d2de82: 
    [08480fbe808800000048895424083cff0f8451ffffff4c89e74983c501ff54c5]
    [00498b16498b46084829d048c1f804480fafc34c39e877b64883c4305b5d415c]
16# 0x00007fa23156b58e: 
    [10faffffc5f97f8560f5ffff0f84fe0000004c89fa4c89ee4c89e7ff95d8f5ff]
    [ff488bbd60f5ffff4885ff740f488bb570f5ffff4829fee806230402488bbd10]
17# 0x00007fa23156b91c: 
    [488d6c24084889efe8374107004989e84c89f94c89f24c89ee4c89e7e8a3deff]
    [ff4889efe8cb4307004883c4104c89e05d415c415d415e415fc34989c4e984ff]
18# 0x00007fa2313c0d46: 
    [efe894e6ffff488b034c8b43084c89e7488b4810488b5020488b7030e899ab1a]
    [00e874b8ffff488dbc24c0020000e847feffff488dbc24d0010000e8caf7ffff]
19# wand::engine::compiler::compiler::linear_order_to_unit_graph(wand::engine::execution::unit_graph_descriptor&&) const in /fast-drive/yakushev/image-watermarking/DWSF/demo_venv/lib/python3.10/site-packages/deepsparse/avx2/libonnxruntime.so.1.15.1
20# wand::engine::compiler::compiler::compile(wand::engine::execution::unit_graph_descriptor&&) const in /fast-drive/yakushev/image-watermarking/DWSF/demo_venv/lib/python3.10/site-packages/deepsparse/avx2/libonnxruntime.so.1.15.1
21# wand::engine::compiler::compiler::compile(wand::engine::execution::graph&&) const in /fast-drive/yakushev/image-watermarking/DWSF/demo_venv/lib/python3.10/site-packages/deepsparse/avx2/libonnxruntime.so.1.15.1
22# wand::engine::compiler::compiler::compile(wand::engine::compute::compute_graph&&) const in /fast-drive/yakushev/image-watermarking/DWSF/demo_venv/lib/python3.10/site-packages/deepsparse/avx2/libonnxruntime.so.1.15.1
23# wand::engine::compiler::compiler::compile(wand::engine::intake::graph&&) const in /fast-drive/yakushev/image-watermarking/DWSF/demo_venv/lib/python3.10/site-packages/deepsparse/avx2/libonnxruntime.so.1.15.1
24# 0x00007fa2305bd927: 
    [ff488bbc24c80000004885ff7405e866b7ebff4c89ea4889de4c89f7e8386d07]
    [0349c7042400000000bff0000000e8361bff02488b0d07350f03488b7040c5f9]
25# 0x00007fa2305c3063: 
    [850801000083400801488dbbb00000004d89e04c89e94c89f24889dee86ca6ff]
    [ff488b45b84989c64885c0741f488b1db9cc0e034885db0f85c00000008b4008]
26# 0x00007fa230604730: 
    [5648488b464c53415450524889fa488bbde0fcffff488985f8fcffffe84fe7fb]
    [ff488bbd08feffff4883c4204885ff7405e83a49e7ff488bbd18fdffff4885ff]
27# 0x00007fa2305cdcc5: 
    [0000488dbc2490000000488d4c24404c8b44241848897c2420488b33e81a5903]
    [00488b4308c5f9efc0c5f96f942490000000c5f97f8424900000004889442430]
28# 0x00007fa2305cf090: 
    [4154c4e1f96eca4989fcc4e3f122c6014883ec104889e6c5f97f0424e82feaff]
    [ff4883c4104c89e0415cc3cccccccccc488b07c5fa6f06c5fa7f00c5fa6f4e10]
29# 0x00007fa230e30a7b: 
    [398002cccc415741564155415455534889fb4883ec38488b0648897c2418ff50]
    [284c8b7308488b1b4c89f04829d84889c248c1f80548c1fa034885c07e6948c1]
30# 0x00007fa230e321ff: 
    [488b742478488dbc24700100004c89f24889c148897c241848890424e860e8ff]
    [ff488b4310488b0bc5f9efc0c5f96fa42470010000488b530848894424104989]

Please email a copy of this stack trace and any additional information to: support@neuralmagic.com
DeepSparse, Copyright 2021-present / Neuralmagic, Inc. version: 1.8.0 (e3778e93) (release) (optimized) (system=avx2, binary=avx2)
Date: 04-11-2025 @ 06:03:54 UTC
OS: Linux teslav100 5.15.0-122-generic #132~20.04.1-Ubuntu SMP Fri Aug 30 15:50:07 UTC 2024
Arch: x86_64
CPU: GenuineIntel
Vendor: Intel
Cores/sockets/threads: [16, 2, 32]
Available cores/sockets/threads: [16, 2, 32]
L1 cache size data/instruction: 32k/32k
L2 cache size: 0.25Mb
L3 cache size: 20Mb
Total memory: 259.79G
Free memory: 35.8419G
Thread: 0x7fa1e7fff700

**Environment**
Include all relevant environment information:
1. OS: Ubuntu 20.04
2. Python version: 3.8.10
3. DeepSparse version or commit hash m,: 1.8.0
4. ML framework version(s): torch 2.4.0
5. Other Python package versions [e.g. SparseML, Sparsify, numpy, ONNX]:
annotated-types==0.7.0
certifi==2025.1.31
charset-normalizer==3.4.1
click==8.1.8
coloredlogs==15.0.1
contourpy==1.3.1
cycler==0.12.1
decorator==5.2.1
deepsparse==1.8.0
filelock==3.18.0
flatbuffers==25.2.10
fonttools==4.57.0
fsspec==2025.3.2
future==1.0.0
geocoder==1.38.1
GPUtil==1.4.0
humanfriendly==10.0
idna==3.10
Jinja2==3.1.6
joblib==1.4.2
kiwisolver==1.4.8
MarkupSafe==3.0.2
matplotlib==3.10.1
merge-args==0.1.5
mpmath==1.3.0
networkx==3.4.2
numpy==1.26.4
nvidia-cublas-cu12==12.4.5.8
nvidia-cuda-cupti-cu12==12.4.127
nvidia-cuda-nvrtc-cu12==12.4.127
nvidia-cuda-runtime-cu12==12.4.127
nvidia-cudnn-cu12==9.1.0.70
nvidia-cufft-cu12==11.2.1.3
nvidia-curand-cu12==10.3.5.147
nvidia-cusolver-cu12==11.6.1.9
nvidia-cusparse-cu12==12.3.1.170
nvidia-cusparselt-cu12==0.6.2
nvidia-nccl-cu12==2.21.5
nvidia-nvjitlink-cu12==12.4.127
nvidia-nvtx-cu12==12.4.127
onnx==1.14.1
onnxruntime==1.21.0
opencv-python==4.11.0.86
packaging==24.2
pandas==2.2.3
pillow==11.1.0
protobuf==3.20.3
psutil==7.0.0
py-machineid==0.7.0
pydantic==2.7.4
pydantic_core==2.18.4
pyparsing==3.2.3
python-dateutil==2.9.0.post0
pytz==2025.2
PyYAML==6.0.2
ratelim==0.1.6
requests==2.32.3
scikit-learn==1.6.1
scipy==1.15.2
six==1.17.0
sparseml==1.8.0
sparsezoo==1.8.1
sympy==1.13.1
threadpoolctl==3.6.0
toposort==1.10
torch==2.6.0
torchvision==0.21.0
tqdm==4.67.1
triton==3.2.0
typing_extensions==4.13.2
tzdata==2025.2
urllib3==2.3.0
6. CPU info - output of deepsparse/src/deepsparse/arch.bin or output of cpu_architecture() as follows:
{'L1_data_cache_size': 32768, 'L1_instruction_cache_size': 32768, 'L2_cache_size': 262144, 'L3_cache_size': 20971520, 'architecture': 'x86_64', 'available_cores_per_socket': 8, 'available_num_cores': 16, 'available_num_hw_threads': 32, 'available_num_numa': 2, 'available_num_sockets': 2, 'available_sockets': 2, 'available_threads_per_core': 2, 'bf16': False, 'cores_per_socket': 8, 'dotprod': False, 'i8mm': False, 'isa': 'avx2', 'num_cores': 16, 'num_hw_threads': 32, 'num_numa': 2, 'num_sockets': 2, 'threads_per_core': 2, 'vbmi': False, 'vbmi2': False, 'vendor': 'GenuineIntel', 'vendor_id': 'Intel', 'vendor_model': 'Intel(R) Xeon(R) CPU E5-2620 v4 @ 2.10GHz', 'vnni': False, 'zen1': False}

**To Reproduce**
Google drive with ONNX model: https://drive.google.com/drive/folders/1U3Vnh2KvGgCDO69WE8D_1G_netcNY2MQ?usp=sharing

to reproduce:
from deepsparse import compile_model
engine = compile_model(path/to/onnx/model)

If needed, i can also send you my UNet model architecture, train/test datasets, traing loop and so on

**Comment by jeanniefinks** [comment]
Per the main [README announcement](https://github.com/neuralmagic/deepsparse/blob/main/README.md), DeepSparse is being deprecated by June 2, 2025. Closing the issue as further development has been suspended; thank you for the inputs and support!
